src/Microsoft.SqlTools.ServiceLayer/LanguageServices/PeekDefinition.cs

```diff
@@ -0,0 +1,210 @@
+//
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+using System;
+using System.IO;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using Microsoft.SqlServer.Management.Smo;
+using Microsoft.SqlServer.Management.SqlParser.Intellisense;
+using Microsoft.SqlTools.ServiceLayer.Connection;
+using Microsoft.SqlTools.ServiceLayer.Workspace.Contracts;
+
+namespace Microsoft.SqlTools.ServiceLayer.LanguageServices
+{
+    /// <summary>
+    /// Peek Definition/ Go to definition implementation
+    /// Script sql objects and write create scripts to file
+    /// </summary>
+    internal class PeekDefinition
+    {
+        private ConnectionInfo connectionInfo;
+        private string tempPath;
+        
+        // Dictionary that holds the script getter for each type
+        private Dictionary<DeclarationType, Func<string, string, StringCollection>> sqlScriptGetters =
```

Comment:
>Func<string, string, StringCollection>> [](start = 44, length = 39)

Minor: we should probably declare this as a delegate (see [https://msdn.microsoft.com/en-us/library/ms173172.aspx](https://msdn.microsoft.com/en-us/library/ms173172.aspx)) to simplify this and shorten the definition. That would look like 
private delegate StringCollection GetScripts(string schemaName, string objectName). 
Also note that in this pattern, we put schemaName first, then objectName (such as table) as this is the standard we use in other codebases for looking up objects.